src/librustc/middle/traits/error_reporting.rs

```diff
@@ -290,7 +290,7 @@ pub fn maybe_report_ambiguity<'a, 'tcx>(infcx: &InferCtxt<'a, 'tcx>,
                     {
                         span_err!(infcx.tcx.sess, obligation.cause.span, E0282,
                                 "unable to infer enough type information about `{}`; \
-                                 type annotations required",
+                                 type annotations or generic parameter binding required",
```

Comment:
with a change like this, you need to update the test suite to reflect the change to the expected error message.

I highly recommend you do a local `make check-stage1-cfail` to catch these, and maybe `make check-stage1-cfail-full` as well ...